services/server/deps.go

```diff
@@ -26,15 +26,22 @@ import (
 	"github.com/percona/pmm-managed/models"
 )
 
+//go:generate mockery -name=checker -case=snake -inpkg -testonly
 //go:generate mockery -name=prometheusService -case=snake -inpkg -testonly
 //go:generate mockery -name=supervisordService -case=snake -inpkg -testonly
 //go:generate mockery -name=telemetryService -case=snake -inpkg -testonly
 
+// Checker interface wraps all services that implements the Check method to report the
+// service health for the Readiness check
+type checker interface {
+	Check(ctx context.Context) error
+}
```

Comment:
Sorry, I think that both interface name and method name are too generic: we already have several checkers like AWSInstanceChecker, and we will have more _checks_ and _checkers_ with a coming feature.

Something like `IsReady(context.Context) bool`?

Reply:
But more importantly, deps.go is for mocking dependencies, not for generic mocks and interfaces.
I assume you want to have `type grafanaClient interface` there like all other dependencies. And you want to accept Grafana client (a real one or mock), not any checker, just like you want to accept Prometheus service (a real one or mock).

Reply:
The idea here was to be able to have a common interface for all services implementing checks. 
Can we name it something like ServiceChecker and keep it here?
If you take a look a couple of lines below, now Prometheus service also implements the `checker` interface. 
We discussed with Nurlan that in the future we might want to add check for other services as well so we implemented checker interface here.

Reply:
@AlekSi so we can provide list of checkers as field of struct you suggested here https://github.com/percona/pmm-managed/pull/360#discussion_r396991827
to loop over them and return a message with non-ready components. In this case we should add extra method to interface `ServiceName() string`
What do you think about that?

Reply:
@AlekSi ^

Reply:
I think it will be easier for me to make a PR to show what I mean. :)

In the meantime, please rename `Check` method to `CheckHealth` everywhere – word "check" becomes overloaded with the introduction of check.percona.con, Security Threat Tool checks, etc.

Reply:
LGTM with recent changes.